tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,569 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import os
+import json
+
 from tests.lib.tools import AgentTestCase
-from azurelinuxagent.ga.policy.policy_engine import PolicyEngine
+from azurelinuxagent.ga.policy.policy_engine import ExtensionPolicyEngine, InvalidPolicyError, \
+    _PolicyEngine, _DEFAULT_ALLOW_LISTED_EXTENSIONS_ONLY, _DEFAULT_SIGNATURE_REQUIRED
 from tests.lib.tools import patch
+from azurelinuxagent.common.protocol.restapi import Extension
+
+TEST_EXTENSION_NAME = "Microsoft.Azure.ActiveDirectory.AADSSHLoginForLinux"
+
+
+class _TestPolicyBase(AgentTestCase):
+    """
+    Define common methods for policy engine test classes.
+    """
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        self.policy_path = os.path.join(self.tmp_dir, "waagent_policy.json")
+
+        # Patch attributes to enable policy feature
+        self.patch_policy_path = patch('azurelinuxagent.common.conf.get_policy_file_path',
+                                       return_value=str(self.policy_path))
+        self.patch_policy_path.start()
+        self.patch_conf_flag = patch('azurelinuxagent.ga.policy.policy_engine.conf.get_extension_policy_enabled',
+                                     return_value=True)
+        self.patch_conf_flag.start()
+
+    def tearDown(self):
+        patch.stopall()
+        AgentTestCase.tearDown(self)
+
+    def _create_policy_file(self, policy):
+        with open(self.policy_path, mode='w') as policy_file:
+            if isinstance(policy, dict):
+                json.dump(policy, policy_file, indent=4)
+            else:
+                policy_file.write(policy)
+            policy_file.flush()
+
+    def _run_test_cases_should_fail_to_parse(self, cases, assert_msg):
+        """
+        Cases should be a list of policies.
+        For each policy in the list, we create a policy file, initialize policy engine, and assert that InvalidPolicyError
+        is raised.
+        """
+        for policy in cases:
+            self._create_policy_file(policy)
+            msg = "invalid policy should not have parsed successfully: {0}. Policy: {1}".format(assert_msg, policy)
+            with self.assertRaises(InvalidPolicyError, msg=msg):
+                _PolicyEngine()
+
+class TestPolicyEngine(_TestPolicyBase):
```

Comment:
nit: there's only one new line before this class

Reply:
fixed